extensions/default/src/Components/DataSourceConfigurationComponent.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { ReactElement, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Icon, useModal } from '@ohif/ui';
+import { ExtensionManager, ServicesManager, Types } from '@ohif/core';
+import DataSourceConfigurationModalComponent from './DataSourceConfigurationModalComponent';
+
+type DataSourceConfigurationComponentProps = {
+  servicesManager: ServicesManager;
+  extensionManager: ExtensionManager;
+};
+
+function DataSourceConfigurationComponent({
+  servicesManager,
+  extensionManager,
+}: DataSourceConfigurationComponentProps): ReactElement {
+  const { t } = useTranslation('DataSourceConfiguration');
+  const { show, hide } = useModal();
+
+  const { customizationService } = servicesManager.services;
+
+  const [configurationAPI, setConfigurationAPI] = useState<
+    Types.BaseDataSourceConfigurationAPI
+  >();
+
+  const [configuredItems, setConfiguredItems] = useState<
+    Array<Types.BaseDataSourceConfigurationAPIItem>
+  >();
+
+  useEffect(() => {
+    let shouldUpdate = true;
```

Comment:
This is odd, why we have shouldUpdate as a `let` and trying to unset it in the return? I expected a useRef at least

Reply:
`shouldUpdate` is a flag that will avoid updating the state in the scenario where the promise resolves AFTER the component has been unmounted. I followed several references for how to do this...

- http://localhost/dicom-web/studies/1.3.12.2.1107.5.8.15.223954.30000023032116563997600001787/series/1.3.12.2.1107.5.8.15.223954.30000023032116563997600001788/metadata
- https://stackoverflow.com/questions/53949393/cant-perform-a-react-state-update-on-an-unmounted-component